src/v2/components/ChannelContents/index.tsx

```diff
@@ -106,21 +104,13 @@ const ChannelContents: React.FC<Props> = WithIsSpiderRequesting<ExtendedProps>(
     )
 
     usePusher({
-      channel: pusherChannel,
+      channelId: channel.id,
+      skip: !isSpiderRequesting,
       onCreated: addBlock,
       onUpdated: updateConnectable,
       parsePayload: parsePayload,
     })
 
-    useEffect(() => {
```

Comment:
All the logic to unsubscribe from a channel is now encapsulated in `usePusher` itself!